google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/it/ITPubSubTest.java

```diff
@@ -403,6 +399,70 @@ public void failed(Subscriber.State from, Throwable failure) {
     topicAdminClient.deleteTopic(topicName);
   }
 
+  @Test
+  public void testPublishSubscribeWithCompression() throws Exception {
+    TopicName topicName =
+        TopicName.newBuilder()
+            .setProject(projectId)
+            .setTopic(formatForTest("testing-compression-topic"))
+            .build();
+    SubscriptionName subscriptionName =
+        SubscriptionName.of(projectId, formatForTest("testing-compression-subscription"));
+
+    topicAdminClient.createTopic(topicName);
+
+    subscriptionAdminClient.createSubscription(
+        getSubscription(subscriptionName, topicName, PushConfig.newBuilder().build(), 10, false));
+
+    final BlockingQueue<Object> receiveQueue = new LinkedBlockingQueue<>();
+    Subscriber subscriber =
+        Subscriber.newBuilder(
+                subscriptionName.toString(),
+                new MessageReceiver() {
+                  @Override
+                  public void receiveMessage(
+                      final PubsubMessage message, final AckReplyConsumer consumer) {
+                    receiveQueue.offer(MessageAndConsumer.create(message, consumer));
+                  }
+                })
+            .build();
+    subscriber.addListener(
+        new Subscriber.Listener() {
+          public void failed(Subscriber.State from, Throwable failure) {
+            receiveQueue.offer(failure);
+          }
+        },
+        MoreExecutors.directExecutor());
+    subscriber.startAsync();
+
+    Publisher publisher = Publisher.newBuilder(topicName).setEnableCompression(true).build();
+
+    String msg1 = generateMessage("msg1", 1000);
+    String msg2 = generateMessage("msg2", 1500);
+    publisher
+        .publish(PubsubMessage.newBuilder().setData(ByteString.copyFromUtf8(msg1)).build())
+        .get();
+    publisher
+        .publish(PubsubMessage.newBuilder().setData(ByteString.copyFromUtf8(msg2)).build())
+        .get();
+    publisher.shutdown();
+    publisher.awaitTermination(1, TimeUnit.MINUTES);
+
+    // Ack the first message.
+    MessageAndConsumer toAck1 = pollQueueMessageAndConsumer(receiveQueue);
+    toAck1.consumer().ack();
+
+    // Ack the second message.
+    MessageAndConsumer toAck2 = pollQueueMessageAndConsumer(receiveQueue);
+    toAck2.consumer().ack();
+
+    assertNotEquals(toAck1.message().getData(), toAck2.message().getData());
```

Comment:
@rajanya-google Can you help me understand the purpose of this test? I see two messages of different sizes getting published using a publisher with no batching configured, so each message is essentially published in its own batch of batch size 1, each message has a size greater than the compression threshold of 10 bytes, and we read these messages and confirm that their data fields aren't equal (what's the point of that)?